### DISCUSSION TITLE
How to correctly patch fonts on Windows

### DISCUSSION THREAD

**Discussion by NoahELE** [ryanoasis/nerd-fonts#1402] [Q&A]
I am trying to patch font on my Windows PC. It seems that the only feasible way is docker.

I ran the following command:

```
docker run --rm -v ~/Desktop/nerd/in:/in:Z -v ~/Desktop/nerd/out:/out:Z nerdfonts/patcher --complete
```

and only got the following result

```
Running with options:
 --complete
```

I put the ttfs in `~/Desktop/nerd/in` folder but get nothing in `~/Desktop/nerd/out` folder.

What is the correct way to do this?

**Comment by Finii** [comment]
The output indicated that Fontforge can not / is not started at all.

I never used Docker on Windows, according to https://docs.docker.com/desktop/install/windows-install/ you need to have WSL installed/activated?

> It seems that the only feasible way is docker.

I would believe the zip patcher should also work (but requires an installed Fontforge).
Let me find some Windows Box.

**Reply by NoahELE** [reply]
Yes, Docker Desktop uses WSL2 backend. I have it installed.

**Reply by Finii** [reply]
The docker script is not very forthcoming with giving users information when/why something failed...

```shell
printf "Running with options:\n%s\n" "$args"

# shellcheck disable=SC2086
for f in /in/*.otf /in/*.ttf /in/*.woff /in/*.eot /in/*.ttc; do [ -f "$f" ] && fontforge -script /nerd/font-patcher -out /out $args "$f"; done

exit 0
```

Maybe we should add some output here, as the failure you described does happen from time to time, always with different reasons.

**Comment by NoahELE** [comment]
I wan't able to get docker to work on my PC, but I managed to patch it with github action.

I create a gist about it https://gist.github.com/NoahELE/92ea3f4b28f75d7117d5bfd9b9fad6d9

**Reply by Finii** [reply]
Great instruction manual, thank you!

**Reply by Finii** [reply]
Just for documentation, this is how I would use it:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/210a3f75-ef57-47e5-a994-fade20997a71)

**Comment by ibanks42** [comment]
I got this to work by using absolute paths instead of relative paths as well as backslashes instead of forward slashes. One of these two may not be needed as I did both at the same time without testing each individually.

``` bash
C:\Users\USERNAME\Desktop\nerd\in
```
instead of
``` bash
~/Desktop/nerd/in
```

Full command:

``` bash
docker run --rm -v C:\Users\USERNAME\Desktop\nerd\in:/in:Z -v C:\Users\USERNAME\Desktop\nerd\out:/out:Z nerdfonts/patcher --complete
```
